nemo/utils/exp_manager.py

```diff
@@ -1122,6 +901,11 @@ def configure_checkpointing(
     if 'mp_rank' in checkpoint_callback.last_model_path or 'tp_rank' in checkpoint_callback.last_model_path:
         checkpoint_callback.last_model_path = uninject_model_parallel_rank(checkpoint_callback.last_model_path)
     trainer.callbacks.append(checkpoint_callback)
+    if create_preemption_callback:
+        ## By default PreemptionCallback handles SIGTERM. To handle other signals pass the signal in the call as below:
```

Comment:
How will the user pass this modification? We need to have a config for the preemption callback which adds this and user can modify. Exp manager bu default only takes trainer object as input and configured everything based on the exp manager config.

Reply:
@titu1994 I have defined the create_preemption_callback arg here: [exp_manager.py#L158](https://github.com/NVIDIA/NeMo/blob/athitten/cluster_preemption/nemo/utils/exp_manager.py#L158) in ExpManagerConfig Dataclass with True as the default value. To make it as False we can just add it in the config under exp_manager, tested it out and it works. Also described the arg in the exp_manager func here: [exp_manager.py#L284](https://github.com/NVIDIA/NeMo/blob/athitten/cluster_preemption/nemo/utils/exp_manager.py#L284). Let me know if this is what you meant, thanks!

Reply:
That enables or disabled the callback. What about if I want to change the list of signals to capture without modifying a single line of code other than the config?

Reply:
Ahh I see this is what you mean, sorry my bad misread it. Why do we want to do a list of signals though ? We can just do SIGTERM by default and we can add a dataclass by name preemption_callback_params in exp_manager with an arg signal carrying a default str value of 'SIGTERM', and if user wants something other than this they can add it in the config yaml file. While calling the [PreemptionCallback class](https://github.com/NVIDIA/NeMo/blob/athitten/cluster_preemption/nemo/utils/exp_manager.py#L907) I'll retrieve the signal value via cfg.preemption_callback_params and pass it to the class.

Reply:
Once we add that, let's merge this